cpp/test/matrix/select.cuh

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <raft/matrix/detail/select_radix.cuh>
+#include <raft/matrix/detail/select_warpsort.cuh>
```

Comment:
We should avoid calling detail APIs in tests. If these are useful for benchmarking and testing, please move them to public APIs. I can't think of any strong reasons why we shouldn't expose them.

Reply:
Closing this conversion to consolidate it in the other thread: https://github.com/rapidsai/raft/pull/1085#discussion_r1053322052